fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -771,7 +771,14 @@ public Subspace indexSubspaceFromMaintainer(@Nonnull Index index) {
     }
 
     @Nonnull
-    public Subspace indexStateSubspace() {
+    public Subspace indexStateSubspace(String indexName, String callingMethod, String targetState) {
+        if (LOGGER.isInfoEnabled()) {
```

Comment:
Why are you adding this here, rather than on `updateIndexState`. Doing it there would remove the unnecessary `indexStateSubspace("<none>", "deleteAllRecords", "<none>")` in `deleteAllRecords`.
To me, that also seems more semantically correct. I'd be surprised if anyone is using this method outside of record-layer-core, but, if they are, there's a good chance they're not using it to change the index state.

Reply:
I wanted to expose anyone who access the state space (directly or via `updateIndexSpace`). But I don't mind changing it.